async.go

```diff
@@ -156,18 +145,16 @@ func (c *Async) CloseChannel() <-chan struct{} {
 // WriteMessage takes a frisbee.Message and some (optional) accompanying content, and queues it up to send asynchronously.
 //
 // If message.ContentLength == 0, then the content array must be nil. Otherwise, it is required that message.ContentLength == len(content).
-func (c *Async) WriteMessage(message *Message, content *[]byte) error {
-	if content != nil && int(message.ContentLength) != len(*content) {
+func (c *Async) WriteMessage(p *packet.Packet) error {
```

Comment:
Using the new `packet.Packet` struct instead of the frisbee.Message + the content pointer, it was way more overhead and made it impossible to keep track of memory usage